### ISSUE TITLE
java.lang.ExceptionInInitializerError: null at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Advapi32Util.java:772)

### ISSUE THREAD

**Issue by lasselindqvist** [oshi/oshi#643]
Version: 3.8.4
OS: Windows 10 Pro that runs as a Jenkins slave

When trying to call new SystemInfo().getHardware().getProcessor() we get the following

```
[main] WARN oshi.util.platform.windows.PerfDataUtil - Failed to add PDH Counter: \Memory\Pages Input/sec, Error code: 0x800007D0
[main] WARN oshi.util.platform.windows.PerfDataUtil - Failed to add PDH Counter: \Paging File(_Total)\% Usage, Error code: 0x800007D0
[main] WARN oshi.util.platform.windows.WmiUtil - COM exception querying Win32_PerfRawData_PerfOS_Memory, which might not be on your system. Will not attempt to query it again. Error was: (HRESULT: 80041003) (puArgErr=):
Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 18.626 sec <<< FAILURE! - in xxx
xxx Time elapsed: 15.935 sec  <<< ERROR!
java.lang.ExceptionInInitializerError: null
	at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Advapi32Util.java:772)
	at oshi.jna.platform.windows.PdhUtil.PdhLookupPerfIndexByEnglishName(PdhUtil.java:99)
	at oshi.hardware.platform.windows.WindowsCentralProcessor.<clinit>(WindowsCentralProcessor.java:68)
	at oshi.hardware.platform.windows.WindowsHardwareAbstractionLayer.getProcessor(WindowsHardwareAbstractionLayer.java:64)
```

Could be related to https://github.com/oshi/oshi/issues/642

Could be related to https://github.com/java-native-access/jna/issues/101

**Comment by dbwiddis** [comment]
The PDH error is 0x800007D0 "Unable to connect to the specified computer, or the computer is offline." The WMI error is 0x80041003, "Access Denied".  

I don't think it's related to either of the links you mentioned but I'm guessing with both WMI and performance counters failing there is some sort of permissions/access error involved here.  Are perfmon counters turned on?  Does the `wmic` command line work?

**Comment by dbwiddis** [comment]
Looking at #642 again perhaps it is a permissions issue.    I'd love to put in a permissions checking warning somewhere, not only on Windows for issues like this but many of the *nix implementations rely on `sudo` commands.

**Comment by dbwiddis** [comment]
> Shame that the library cannot tell about missing permissions with a checked exception.

The log warnings are checked exceptions individually for the PDH and, if it fails, WMI. That's how it pointed to those access denied error codes.   What's missing is catching the error failing to read the registry -- now that I reread the actual exception this is likely the root cause of #642.   Do you have more of that original stack trace, the next lines "Caused by" should show the exception we should catch (probably a Win32Exception in Advapi32Util?)

**Comment by lasselindqvist** [comment]
Win32Exception in Advapi32Util is indeed the thrown exception. It is thrown here: https://github.com/java-native-access/jna/blob/4.5.2/contrib/platform/src/com/sun/jna/platform/win32/Advapi32Util.java#L772

**Comment by dbwiddis** [comment]
Actually... looks like I already fixed this when reported in #633 which was fixed in #636 and should be in SNAPSHOT.  Can you test that (for this error, at least).

So this is related to #642 (permissions issue) but a completely different location/error.   It's a good spot to potentially have a log warning about registry permissions as well.
